WIP: Fix panel.margin with positive values (#180)

ANAMASGARD opened the pull request:
FIXES #180

## Current Status

Tests failing (6/22) - this commit demonstrates the bug exists.

## Problem Identified

pt.to.lines() returns constant 0.25 for all inputs instead of converting the actual value.

Current behavior:
- grid::unit(2, "lines") returns 0.25 (expected: 2)
- grid::unit(1, "cm") returns 0.25 (expected: ~2.86)
- grid::unit(0, "lines") returns 0.25 (expected: 0)

## Test Output

<img width="2725" height="1539" alt="Screenshot From 2025-12-23 09-58-26" src="https://github.com/user-attachments/assets/b7cbbd15-f884-490f-a15e-06221d8bb140" />

ANAMASGARD committed:
test: Add failing tests for panel.margin with positive values

Tests demonstrate:
- pt.to.lines() returns 0.25 for all inputs (should vary)
- Zero values incorrectly return 0.25 instead of 0
- Positive values (2 lines, 1 cm) return 0.25 instead of correct conversion .

ANAMASGARD committed:
test: add tests for positive panel.margin values

Issue #180 asked for test coverage of positive panel.margin values.
Added tests for lines, cm, and pt units. All tests pass - the
feature already works, just needed proper test coverage.

Fixes #180

ANAMASGARD commented:
Sir @tdhock The R_coverage check is failing on an unrelated test 
(`test-compiler-ghpages.R:63`) that expects 1 TSV file but finds 2.

This appears unrelated to panel.margin since my PR only adds test coverage 
and doesn't modify any compilation or TSV generation code. 

Sir can you please review the PR and guide me on what should I be doing next .

tdhock reviewed (changes requested):
please add a renderer test

ANAMASGARD committed:
Added tests for positive panel.margin (fixes #180)

ANAMASGARD committed:
Merge branch 'master' into fix-panel-margin-positive-values

ANAMASGARD committed:
Merge branch 'master' into fix-panel-margin-positive-values

ANAMASGARD committed:
Merge branch 'master' into fix-panel-margin-positive-values

ANAMASGARD committed:
test(#180): cover facet_wrap and horizontal facet_grid panel.margin

tdhock committed:
Merge branch 'master' into fix-panel-margin-positive-values

ANAMASGARD committed:
Merge branch 'master' into fix-panel-margin-positive-values

ANAMASGARD committed:
Merge branch 'master' into fix-panel-margin-positive-values

ANAMASGARD committed:
Merge branch 'master' into fix-panel-margin-positive-values

ANAMASGARD committed:
Merge branch 'master' into fix-panel-margin-positive-values

ANAMASGARD committed:
fix: resolve duplicate Version in DESCRIPTION for CI (PR#286)

Merge left two Version fields (2026.4.17 and 2026.4.28), which caused
pak::lockfile_create to fail before tests ran. Use single Version
2026.5.29 and update NEWS accordingly.

tdhock commented on NEWS.md:
```diff
@@ -1,3 +1,7 @@
+# Changes in version 2026.5.29 (PR#286)
+
+- Added regression tests for `panel.margin` with positive values (issue #180), covering vertical/horizontal `facet_grid` and `facet_wrap`.
```
thank yourself

tdhock replied:
also please give an example here. and please explain what units are supported? Lines? pixels?

ANAMASGARD committed:
docs(NEWS): address tdhock review for panel.margin (PR#286, #180)

Thanks @ANAMASGARD; document supported units (lines recommended).
Add inst/examples/panel-margin-issue-180.R for horizontal facet_grid
0 vs 2 lines (screenshots posted on PR separately).

tdhock commented on NEWS.md:
```diff
@@ -1,3 +1,19 @@
+  "lines")` → `n` line-heights in the browser). **`"pt"`** / **`"points"`** are
+  converted with the ggplot2 default scale (~5.5 pt → 0.25 lines). Other units
+  (`"cm"`, `"mm"`, etc.) pass through as a numeric value interpreted as a line
+  count, not a physical conversion. Pixels are not used directly.
```
please delete or clarify.
if only lines are supported then we should error for other units?

ANAMASGARD committed:
Removed supported-units paragraph per review
Drop the long units block ; kept tests and
example pointer for issue.

tdhock commented on tests/testthat/test-renderer1-panel-margin.R:
```diff
@@ -0,0 +1,180 @@
+  i2 <- abs(xs - ux[2]) < tol
+  left2 <- min(xs[i2])
+  left2 - right1
+}
```
please revise to avoid repeated logic

ANAMASGARD replied:
Refactored test-renderer1-panel-margin.R per review - replaced eight helpers with panel_attr() and gap_between_positions(). Locally re ran test it passes

ANAMASGARD committed:
refactor(tests): replace 8 helpers with panel_attr/gap_between_positions (#286)
Address tdhock review: deduplicate panel.margin renderer test helpers.
One panel_attr() and one gap_between_positions() (with return() early exit).

ANAMASGARD committed:
Merge branch 'master' into fix-panel-margin-positive-values

tdhock commented on tests/testthat/test-panel-margin-positive.R:
```diff
@@ -0,0 +1,84 @@
+test_that("pt.to.lines handles positive cm unit correctly", {
+  cm_value <- grid::unit(0.5, "cm")
+  converted <- pt.to.lines(cm_value)
+  expect_true(is.numeric(converted))
```
use expect_is, or remove

tdhock replied:
same for below

tdhock commented on NEWS.md:
```diff
@@ -1,3 +1,12 @@
+# Changes in version 2026.5.29 (PR#286)
+
+- Added regression tests for `panel.margin` with positive values (issue #180),
+  covering vertical/horizontal `facet_grid` and `facet_wrap`. Thanks @ANAMASGARD.
```
please describe new features, not new tests

tdhock reviewed (changes requested):
please revise

ANAMASGARD committed:
Merge branch 'master' into fix-panel-margin-positive-values

ANAMASGARD committed:
test(#180): simplify assertions and revise NEWS

Remove redundant is.numeric() checks per review. Describe panel.margin
support in NEWS instead of listing added tests.

tdhock reviewed (approved):
great thanks
